File: libgomp/testsuite/libgomp.fortran/target-foffload_disable.f
!     { dg-additional-options "-foffload=disable" }

      PROGRAM MAIN
      IMPLICIT NONE

      INCLUDE "omp_lib.h"

      IF (.NOT. OMP_IS_INITIAL_DEVICE()) CALL ABORT
!$OMP TARGET
      IF (.NOT. OMP_IS_INITIAL_DEVICE()) CALL ABORT
!$OMP END TARGET
      IF (.NOT. OMP_IS_INITIAL_DEVICE()) CALL ABORT

      END

File: libgomp/testsuite/libgomp.c/target-foffload_disable.c
/* { dg-additional-options "-foffload=disable" } */

#include <omp.h>

int main()
{
  if (!omp_is_initial_device())
    __builtin_abort();
#pragma omp target
  {
    if (!omp_is_initial_device())
      __builtin_abort();
  }
  if (!omp_is_initial_device())
    __builtin_abort();

  return 0;
}
